packages/generator/src/service-name-formatter.ts

```diff
@@ -24,13 +38,26 @@ export class ServiceNameFormatter {
     [entitySetOrComplexTypeName: string]: UniqueNameFinder;
   } = {};
 
+  constructor();
+  /**
+   * @deprecated since version 1.25.0. The name formatters for the sets, types and function imports are initialized lazy now so there is no need to pass the names beforehand.
+   * Use the argument free constructor instead.
+   * @param entitySetNames The entity set names.
+   * @param complexTypeNames The complex type names.
+   * @param functionImportNames Then function import names.
+   */
   constructor(
     entitySetNames: string[],
     complexTypeNames: string[],
     functionImportNames: string[]
+  );
+  constructor(
+    entitySetNames?: string[],
+    complexTypeNames?: string[],
+    functionImportNames?: string[]
   ) {
     // Here we assume that entitysets and complextypes cannot have the same original name
-    [...entitySetNames, ...complexTypeNames].forEach(
+    [...(entitySetNames || []), ...(complexTypeNames || [])].forEach(
```

Comment:
Togher with the suggestion above:
```suggestion
    [...entitySetNames, ...complexTypeNames].forEach(
```